network/network.go

```diff
@@ -33,10 +34,10 @@ const (
 	PublicNetwork
 )
 
-// Network represents the network layer of the node. It allows processes that
-// work across the peer-to-peer network to register themselves as an engine with
-// a unique engine ID. The returned conduit allows the process to communicate to
-// the same engine on other nodes across the network in a network-agnostic way.
+// Network is one of the networking layer interfaces in Flow (i.e., Network, Adapter, and Middleware). It represents the interface that networking layer
+// offers to the Flow protocol layer, i.e., engines. It is responsible for creating conduits through which engines
+// can send and receive messages to and from other engines on the network, as well as registering other services
+// such as BlobService and PingService.
 type Network interface {
```

Comment:
```suggestion
type NetworkEngine interface {
```

Reply:
Per our conversation, renamed `Network` to `EngineRegistry`: https://github.com/onflow/flow-go/pull/4664/commits/24cbe582986ad9e024df68cba7a9ffbf43899bbb